intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/problems/LevelRemappingPluginCreationResultResolver.kt

```diff
@@ -58,7 +59,13 @@ class LevelRemappingPluginCreationResultResolver(private val delegatedResolver:
   }
 
   private fun remapPluginProblemLevel(pluginProblem: PluginProblem): PluginProblem?{
-    return when (val remappedLevel = remappedLevel[pluginProblem::class]) {
+    val problem = if (unwrapRemappedProblems && pluginProblem is ReclassifiedPluginProblem)  {
```

Comment:
Needing to cast is not nice. Similar code here PluginProblem.isInvalidDescriptorProblem()

Reply:
The `ReclassifiedPluginProblem` is intentionally a subclass of a `PluginProblem`: the similar approach as in `Exception` and exception wrapping. 

In addition, reclassified plugin problems is a separate branch in inheritance hierarchy.

I don't know how to implement this without instance checking.

Reply:
Refactored to use extension functions.